### PR TITLE
Added Bootstrap Modals for Uploading Dataset and Model in Dashboard2

### PR DESCRIPTION
**Pull request by Navnoorsinghmahal**
## 📜 Description

This pull request implements a **Bootstrap Modal** feature for the model upload instructions in the Dashboard. The modal provides clear guidelines for users to upload models in the correct format and size, enhancing user experience and ensuring smoother interactions.

---

## ✅ Changes Made

- Added a **Bootstrap Modal** to the Dashboard for displaying upload instructions.
- Included details about:
  - **File Format**: `.pkl` files only.
  - **File Size**: Less than 1 MB.
- Integrated buttons in the modal for:
  - **Upload Model**: Triggers the file input.
  - **Close**: Closes the modal.
- Styled the modal using inline CSS for customization.
- Adjusted modal size to be compact for a better fit within the UI.

---

## ✅ Requirements Addressed

- Provide users with clear instructions on acceptable file formats and size limits.
- Ensure the modal is reusable and adheres to Bootstrap conventions.
- Maintain a clean and user-friendly interface.

---

## 📋 Acceptance Criteria

- [x] A Bootstrap Modal is present and opens when triggered.
- [x] The modal displays clear upload instructions.
- [x] Buttons in the modal function as expected:
  - "Upload Model" triggers file input and closes the modal.
  - "Close" simply closes the modal.
- [x] The modal design is compact and visually aligns with the rest of the Dashboard.

### CODE REVIEW

**Comment by armaganngul** [comment]
In the joyride thing, when user clicks on the upload button this modal comes on top of the joyride gray background right? just to make sure

**Review by armaganngul — COMMENTED** [review]
Maybe we could have done this with a Modal component class like for Dashboard1? that would make it more CLEAN i think. but looks good

File: frontend/src/pages/ModelTester/ControlButtons2/ControlButtons2.jsx [hunk 1]
```diff
@@ -211,6 +218,58 @@ const ControlButton2 = ({ setUploadedFiles }) => {
...
+                                        padding: "0.25rem 0.5rem",  // Smaller padding
+                                        borderRadius: "0.2rem"  // Optional: round corners slightly
+                                    }}
+                                    onClick={closeModal}
+                                >
+                                    &times;
+                                </button>
+                            </div>
+                            <div className="modal-body">
+                                <p><strong>File format:</strong> The file must be in <code>.pkl</code> format.</p>
```
[the remark this answers is not in the document]
Comment (Navnoorsinghmahal) [reply frontend/src/pages/ModelTester/ControlButtons2/ControlButtons2.jsx]: Added the changes

File: frontend/src/pages/ModelTester/ControlButtons2/ControlButtons2.jsx [hunk 2]
```diff
@@ -211,6 +218,58 @@ const ControlButton2 = ({ setUploadedFiles }) => {
...
+                                <p><strong>File size:</strong> The file size must be less than 1 MB.</p>
```
[the remark this answers is not in the document]
Comment (Navnoorsinghmahal) [reply frontend/src/pages/ModelTester/ControlButtons2/ControlButtons2.jsx]: Done with the change

**Review by Jai0212 — COMMENTED** [review]
Great! Seems correct now

**Pull request merged**